### ISSUE TITLE
Home Occupied switch is not working consistently

### ISSUE THREAD

**Issue by johnslemmer** [chrisjshull/homebridge-nest#331]
## Description

The Home Occupied switch is being very finicky.  Most of the time (sometimes it works right away) when I tap it to change its state is flashes to the opposite state and the switches back.  Usually takes a couple tries before my desired state takes.  This is a bummer becuase I manipulate this switch in some automations, but because the automations only try to change something once then it doesn't hold.

I started noticing this recently.  But couldn't give you an exact date.  Not sure if it is iOS 14 related, but that doesn't seem to be the cause as I get a similar behavior from the homebridge UI.

See attached logs below.  You can see at `6:57:22 PM` I click to turn the Home Occupied switch to on and it only stays on for an instant and then switches back to off/false.  And right after I try again at `6:57:46 PM` but then for some reason works at `6:57:58 PM`.

Any ideas?  Let me know how we can debug this further.

## To Reproduce

Just click the Home Occupied button (in either iOS Home App or in the hombridge web-ui).

## Expected behavior

That the Home Occupied button should work consistently.

## Log

```
[11/4/2020, 6:55:38 PM] [HB Supervisor] OS: Linux 4.15.0-112-generic x64
[11/4/2020, 6:55:38 PM] [HB Supervisor] Node.js v12.18.3 /usr/local/bin/node
[11/4/2020, 6:55:38 PM] [HB Supervisor] Homebridge Path: /usr/local/lib/node_modules/homebridge/bin/homebridge
[11/4/2020, 6:55:38 PM] [HB Supervisor] UI Path: /usr/local/lib/node_modules/homebridge-config-ui-x/dist/bin/standalone.js
[11/4/2020, 6:55:38 PM] [HB Supervisor] Starting Homebridge with extra flags: -I -P /homebridge/node_modules -D
[11/4/2020, 6:55:38 PM] [HB Supervisor] Started Homebridge v1.1.6 with PID: 8056
[11/4/2020, 6:55:39 PM] Loaded config.json with 0 accessories and 2 platforms.
[11/4/2020, 6:55:39 PM] ---
[11/4/2020, 6:55:39 PM] Loaded plugin: homebridge-dummy@0.4.1
[11/4/2020, 6:55:39 PM] Registering accessory 'homebridge-dummy.DummySwitch'
[11/4/2020, 6:55:39 PM] ---
[11/4/2020, 6:55:39 PM] Loaded plugin: homebridge-nest@4.4.15
[11/4/2020, 6:55:39 PM] Registering platform 'homebridge-nest.Nest'
[11/4/2020, 6:55:39 PM] ---
[11/4/2020, 6:55:39 PM] Loaded plugin: homebridge-config-ui-x@4.31.0
[11/4/2020, 6:55:39 PM] Registering platform 'homebridge-config-ui-x.config'
[11/4/2020, 6:55:39 PM] ---
[11/4/2020, 6:55:39 PM] Loading 2 platforms...
[11/4/2020, 6:55:39 PM] [Nest] Initializing Nest platform...
[11/4/2020, 6:55:39 PM] [Config] Initializing config platform...
[11/4/2020, 6:55:39 PM] [Config] Running in Service Mode
[11/4/2020, 6:55:39 PM] [Nest] Fetching Nest devices.
[11/4/2020, 6:55:39 PM] [Nest] Authenticating via Google.
Setup Payload:
XXXXXXXXXXXXXXX
Enter this code with your HomeKit app on your iOS device to pair with Homebridge:
XXXXXXXXXXXXXXX
[11/4/2020, 6:55:39 PM] Homebridge is running on port 51054.
[11/4/2020, 6:55:41 PM] [Nest] Authentication successful.
[11/4/2020, 6:55:44 PM] [Homebridge UI] Homebridge Config UI X v4.31.0 is listening on :: port 8124
[11/4/2020, 6:55:43 PM] [Nest] initing thermostat "Dining Room Thermostat": deviceId: XXXXXXXXXXXXXXX structureId: XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
[11/4/2020, 6:55:43 PM] [Nest] Temperature unit for Dining Room Thermostat is: Fahrenheit
[11/4/2020, 6:55:43 PM] [Nest] Current temperature for Dining Room Thermostat is: 20.429 °C / 69 °F
[11/4/2020, 6:55:43 PM] [Nest] Current humidity for Dining Room Thermostat is: 49%
[11/4/2020, 6:55:43 PM] [Nest] Target temperature for Dining Room Thermostat is: 20 °C / 68 °F
[11/4/2020, 6:55:43 PM] [Nest] Target heating/cooling state for Dining Room Thermostat is: Heat
[11/4/2020, 6:55:43 PM] [Nest] Cooling threshold temperature for Dining Room Thermostat is: 26.7 °C / 80 °F
[11/4/2020, 6:55:43 PM] [Nest] Heating threshold temperature for Dining Room Thermostat is: 10 °C / 50 °F
[11/4/2020, 6:55:43 PM] [Nest] Online status for Dining Room Thermostat is: true
[11/4/2020, 6:55:43 PM] [Nest] Sunlight Correction Enabled for Dining Room Thermostat is: true
[11/4/2020, 6:55:43 PM] [Nest] initing home_away_sensor "Home Occupied": deviceId: XXXXXXXXXXXXXXXXXXXXXXXXX structureId: XXXXXXXXXXXXXXXXXXX
[11/4/2020, 6:55:43 PM] [Nest] Home Occupied for Home Occupied is: true
[11/4/2020, 6:56:31 PM] [Nest] Home Occupied for Home Occupied is: false
[11/4/2020, 6:56:31 PM] [Nest] Home Occupied for Home Occupied is: true
[11/4/2020, 6:56:36 PM] [Nest] Home Occupied for Home Occupied is: false
[11/4/2020, 6:56:38 PM] [Nest] Current humidity for Dining Room Thermostat is: 47%
[11/4/2020, 6:56:38 PM] [Nest] Target temperature for Dining Room Thermostat is: 20.429 °C / 69 °F
[11/4/2020, 6:56:38 PM] [Nest] Target heating/cooling state for Dining Room Thermostat is: Auto
[11/4/2020, 6:56:38 PM] [Nest] Cooling threshold temperature for Dining Room Thermostat is: 29.444 °C / 85 °F
[11/4/2020, 6:56:38 PM] [Nest] Eco Mode for Dining Room Thermostat is: true
[11/4/2020, 6:57:22 PM] [Nest] Home Occupied for Home Occupied is: true
[11/4/2020, 6:57:22 PM] [Nest] Home Occupied for Home Occupied is: false
[11/4/2020, 6:57:46 PM] [Nest] Home Occupied for Home Occupied is: true
[11/4/2020, 6:57:46 PM] [Nest] Home Occupied for Home Occupied is: false
[11/4/2020, 6:57:58 PM] [Nest] Home Occupied for Home Occupied is: true
[11/4/2020, 6:58:00 PM] [Nest] Target temperature for Dining Room Thermostat is: 20 °C / 68 °F
[11/4/2020, 6:58:00 PM] [Nest] Target heating/cooling state for Dining Room Thermostat is: Heat
[11/4/2020, 6:58:00 PM] [Nest] Cooling threshold temperature for Dining Room Thermostat is: 26.7 °C / 80 °F
[11/4/2020, 6:58:00 PM] [Nest] Eco Mode for Dining Room Thermostat is: false
```

**Comment by adriancable** [comment]
@johnslemmer - we can reproduce this. It seems to be something on the Nest service side. We have heard recently of issues of thermostats not going into Eco mode when you're Away (nothing at all to do with Homebridge or plug-ins) which is probably the same issue.

We're looking into workarounds and will update you.

**Comment by djrams10** [comment]
thanks, I have the same problem

**Comment by adriancable** [comment]
I've just published version 4.5.0, which should address this - give it a go and let me know how you get on.
